fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Optimizer.java

```diff
@@ -301,8 +301,11 @@ private OptExpression logicalRuleRewrite(OptExpression tree, TaskContext rootTas
             }
         }
 
-        tree = new MaterializedViewRule().transform(tree, context).get(0);
-        deriveLogicalProperty(tree);
+        if (sessionVariable.isEnableSyncMaterializedViewRewrite()) {
+            // Add a config to decide whether to rewrite sync mv.
+            tree = new MaterializedViewRule().transform(tree, context).get(0);
```

Comment:
MaterializedViewRule should check and respect the hint.

Reply:
fix it.